src/lib/y2storage/proposal_settings.rb

```diff
@@ -53,11 +99,43 @@ def initialize
       @use_separate_home        = true
       @home_filesystem_type     = Filesystems::Type::XFS
       @enlarge_swap_for_suspend = false
+      @resize_windows           = true
+      @windows_delete_mode      = :ondemand
+      @linux_delete_mode        = :ondemand
+      @other_delete_mode        = :ondemand
     end
 
     def use_encryption
       !encryption_password.nil?
     end
+
+    # Whether the settings disable deletion of a given type of partitions
+    #
+    # @see #windows_delete_mode
+    # @see #linux_delete_mode
+    # @see #other_delete_mode
+    #
+    # @param type [#to_s] :linux, :windows or :other
+    # @return [Boolean]
+    def delete_forbidden(type)
+      send(:"#{type}_delete_mode").to_sym == :none
```

Comment:
and why to_sym is needed? From documentation it looks like it should return always symbol